src/services/composite-keys-manager.js

```diff
@@ -1,54 +1,83 @@
 import Operators from '../utils/operators';
 
-function CompositeKeysManager(model, schema, record) {
-  const GLUE = '|';
-
-  this.getPrimaryKeyValues = function getPrimaryKeyValues(recordId) {
-    const primaryKeyValues = recordId.split(GLUE);
-
-    // NOTICE: Prevent liana to crash when a composite primary keys is null,
-    //         this behaviour should be avoid instead of fixed.
-    primaryKeyValues.forEach((key, index) => {
-      if (key === 'null') { primaryKeyValues[index] = null; }
-    });
-
-    return primaryKeyValues;
-  };
-
-  this.getRecordsConditions = function getRecordsConditions(recordsIds, options) {
-    const { OR } = Operators.getInstance(options);
-    return { [OR]: recordsIds.map((recordId) => this.getRecordConditions(recordId)) };
-  };
-
-  this.getRecordConditions = function getRecordConditions(recordId) {
-    const where = {};
-    const primaryKeyValues = this.getPrimaryKeyValues(recordId);
-    if (primaryKeyValues.length === _.keys(model.primaryKeys).length) {
-      _.keys(model.primaryKeys).forEach((primaryKey, index) => {
-        where[primaryKey] = primaryKeyValues[index];
+/**
+ * This helper class allows abstracting away the complexity
+ * of using collection which have composite primary keys.
+ */
+class CompositeKeysManager {
+  static _GLUE = '|'
+
+  constructor(model) {
+    this._primaryKeys = _.keys(model.primaryKeys);
+    this._Sequelize = model.sequelize.constructor;
+  }
+
+  /** Build sequelize where condition from a list of packed recordIds */
+  getRecordsConditions(recordIds) {
+    if (recordIds.length <= 0) {
+      return this._Sequelize.literal('(0=1)');
+    }
+
+    switch (this._primaryKeys.length) {
+      case 0:
+        throw new Error('No primary key was found');
+
+      case 1:
+        return this._getRecordsConditionsSimple(recordIds);
+
+      default:
+        return this._getRecordsConditionComposite(recordIds);
+    }
+  }
+
+  /* Annotate records with their packed primary key */
+  annotateRecords(records) {
+    if (this._primaryKeys.length > 1) {
+      records.forEach((record) => {
+        record.forestCompositePrimary = this._createCompositePrimary(record);
       });
     }
-    return where;
-  };
-
-  this.createCompositePrimary = function createCompositePrimary() {
-    let compositePrimary = '';
-
-    _.keys(model.primaryKeys).forEach((primaryKey, index) => {
-      // NOTICE: Prevent liana to crash when a composite primary keys is null,
-      //         this behaviour should be avoid instead of fixed.
-      if (record[primaryKey] === null) {
-        record[primaryKey] = 'null';
-      }
-      if (index === 0) {
-        compositePrimary = record[primaryKey];
-      } else {
-        compositePrimary = compositePrimary + GLUE + record[primaryKey];
-      }
-    });
-    return compositePrimary;
-  };
+  }
+
+  _getRecordsConditionsSimple(recordIds) {
+    return { [this._primaryKeys[0]]: recordIds.length === 1 ? recordIds[0] : recordIds };
```

Comment:
```suggestion
    return { [this._primaryKeys[0]]: recordIds };
```

No need to transform it both works

Reply:
Removing extra arrays and operators in the generated query options makes using the debugger much easier when working on those files so I prefer to keep it this way.